### DISCUSSION TITLE
Keyboard naviagtion: move panes (tabs)

### DISCUSSION THREAD

**Discussion by lucaspar** [zed-industries/zed#14121] [Feature Requests]
Is there currently a way to move tabs/panes around with the keyboard in a similar way we can do with the mouse by dragging the tab around?

A VS Code equivalent to `workbench.action.moveEditorTo[Left/Right/Above/Below]Group`.

**Comment by lucaspar** [comment]
Also, what's Zed terminology for pane "groups"? Or is a pane only the tab that is active?

**Comment by tepavcevic** [comment]
Take a look [here](https://zed.dev/docs/key-bindings#workspace), there are keybindings `ctrl-k shift-right/left` that will allow you to swap panes left <-> right or up <-> down.

**Reply by lucaspar** [reply]
Thanks; I guess that's the closest behavior we can get right now.

One key difference from VS Code's implementation is that, there, this is a proper "move" operation instead of swap: when there's no pane to swap in a direction, the command behaves like Zed's "split left/right/up/down".

The problem of not having a "move" operation is that splitting duplicates the active pane in these cases where there's no pane to swap for.

Maybe this behavior could be added in the future, as it also replicates mouse actions more consistently.

**Reply by tepavcevic** [reply]
Yes, that is one thing I am missing from JetBrains IDE's as well, moving current tab to the pane on the side and the ability to bring it back to the original pane.

**Comment by btrazzini** [comment]
I often use keyboard shortcuts to organize the position of each open tab. This is especially useful when I need to compare file 1 with file 2, positioning them side by side. Being able to change the position of tabs via the keyboard significantly boosts productivity. 

In VS Code, I have configured the shortcuts as follows:

```json
{
  "key": "ctrl+shift+left",
  "command": "workbench.action.moveEditorLeftInGroup"
},
{
  "key": "cmd+k shift+cmd+left",
  "command": "-workbench.action.moveEditorLeftInGroup"
},
{
  "key": "ctrl+shift+right",
  "command": "workbench.action.moveEditorRightInGroup"
},
{
  "key": "cmd+k shift+cmd+right",
  "command": "-workbench.action.moveEditorRightInGroup"
}
```

Implementing similar keyboard shortcuts in Zed Editor would greatly enhance the workflow for users who rely on efficient tab management.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support

**Comment by lucaspar** [comment]
It seems we can now do this, which behaves exactly as I wanted when I opened this discussion.

```json
"ctrl-k ctrl-left": [
    "workspace::MoveItemToPaneInDirection",
    { "direction": "left" },
],
"ctrl-k ctrl-right": [
    "workspace::MoveItemToPaneInDirection",
    { "direction": "right" },
],
"ctrl-k ctrl-up": [
    "workspace::MoveItemToPaneInDirection",
    { "direction": "up" },
],
"ctrl-k ctrl-down": [
    "workspace::MoveItemToPaneInDirection",
    { "direction": "down" },
],
```

**Comment by lucaspar** [comment]
@JosephTLyons that's unfortunate, because not everyone wants to use Discord, and finding information there in the middle of so much noise is... challenging.

**Comment by J1337** [comment]
@lucaspar 

Thanks for the feedback.

Discord gives us much better capabilities for maintaining a growing community, and it’s where the Zed support team is actively monitoring questions. We are a small team, so focusing our efforts in one place lets us respond consistently, rather than splitting attention across multiple platforms and leaving questions sitting open for long periods of time.
